osutil/vfs/mount.go

```diff
@@ -0,0 +1,166 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package vfs
+
+import (
+	"io/fs"
+)
+
+// Mount attaches the given file system to given mount point.
+//
+// Mount attaches a mount entry to the node represented by the mount point. The
+// mount point must be an existing directory in the VFS. If the mount point
+// contains other mount entries they are shadowed until the new entry is
+// unmounted.
+func (v *VFS) Mount(fsFS fs.StatFS, mountPoint string) error {
+	// Hold lock throughout the function as we need to consistently mutate m.mounts at the end.
+	v.mu.Lock()
+	defer v.mu.Unlock()
+
+	// Find the mount dominating the mount point.
+	_, dom, _, fsPath := v.dom(mountPoint)
```

Comment:
Looking at implementation for `mount.isDom`, shouldn't the mountPoint be validated to not end in a `/`?

Reply:
I need to think a little and perhaps add some more tests.

Reply:
I now reject those and I've added test coverage for this.